### PR TITLE
[LEARNER-326] update analytics to support multi line basket

### PR DESCRIPTION
**Pull request by marjev**
Reviewers:
@edx/helio 

JIRA ticket:
[update analytics to support multi line basket](https://openedx.atlassian.net/browse/LEARNER-326)

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @marjev! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fecommerce&number=1238"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

File: ecommerce/templates/oscar/basket/partials/client_side_checkout_basket.html [hunk 1]
```diff
@@ -113,7 +113,7 @@ <h2 id="payment-method-region" class="title">{% trans "select payment method" %}
                                 data-track-event="edx.bi.ecommerce.basket.payment_selected"
                                 data-track-category="checkout"
                                 data-processor-name="paypal"
-                                data-course-id="{{ course_key }}"
+                                data-course-ids="{{ course_keys }}"
```
Comment (clintonb) [inline ecommerce/templates/oscar/basket/partials/client_side_checkout_basket.html]: Nothing in our code base actually uses this attribute. All of this should go away.

**Review by clintonb — CHANGES_REQUESTED** [review]

File: ecommerce/templates/oscar/basket/partials/client_side_checkout_basket.html (hunk 1 above)
Comment (clintonb) [reply ecommerce/templates/oscar/basket/partials/client_side_checkout_basket.html]: "All of this" includes the data attributes and the context variable.

**Comment by marjev** [comment]
@clintonb I removed unused code.

File: ecommerce/templates/oscar/basket/partials/client_side_checkout_basket.html [hunk 2]
```diff
@@ -110,12 +110,7 @@ <h2 id="payment-method-region" class="title">{% trans "select payment method" %}
                     </div>
                     <div id="payment-processor" class="payment-buttons col-md-3 col-sm-5 col-xs-6" data-basket-id="{{ basket.id }}">
                         {# Translators: Do NOT translate the name PayPal. #}
-                        <button data-track-type="click"
-                                data-track-event="edx.bi.ecommerce.basket.payment_selected"
-                                data-track-category="checkout"
-                                data-processor-name="paypal"
-                                data-course-id="{{ course_key }}"
```
Comment (clintonb) [inline ecommerce/templates/oscar/basket/partials/client_side_checkout_basket.html]: The tracking is used. It's only this data attribute that is not used.

File: ecommerce/templates/oscar/basket/partials/client_side_checkout_basket.html [hunk 3]
```diff
@@ -196,21 +191,11 @@ <h2 id="billing-information-region" class="title">{% trans "billing information"
                 <div class="payment-button col-sm-6 col-sm-offset-6 col-xs-12">
                 {% if free_basket %}
                     <a href="{% url 'checkout:free-checkout' %}"
-                       data-track-type="click"
-                       data-track-event="edx.bi.ecommerce.basket.free_checkout"
-                       data-track-category="checkout"
-                       data-course-id="{{ course_key }}"
```
Comment (clintonb) [inline ecommerce/templates/oscar/basket/partials/client_side_checkout_basket.html]: This is the only line that should be removed.

File: ecommerce/templates/oscar/basket/partials/client_side_checkout_basket.html [hunk 4]
```diff
@@ -196,21 +191,11 @@ <h2 id="billing-information-region" class="title">{% trans "billing information"
...
                        class="btn btn-primary btn-large col-sm-12 col-xs-12">
                         {% trans "Place Order" %}
                     </a>
                 {% else %}
-                    <button id="payment-button" type="submit" class="btn btn-primary btn-large col-sm-12 col-xs-12"
-                            data-track-type="click"
-                            data-track-event="edx.bi.ecommerce.basket.payment_selected"
-                            data-track-category="checkout"
-                            data-processor-name="{{ client_side_payment_processor.NAME }}"
-                            data-course-id="{{ course_key }}"
```
Comment (clintonb) [inline ecommerce/templates/oscar/basket/partials/client_side_checkout_basket.html]: This is the only line that should be removed.

**Review by clintonb — CHANGES_REQUESTED** [review]
Whoa. *Tracking* is used. We simply don't use the `data-course-id` attribute.

File: ecommerce/templates/oscar/basket/partials/client_side_checkout_basket.html (hunk 4 above)
Comment (vkaracic) [reply ecommerce/templates/oscar/basket/partials/client_side_checkout_basket.html]: Are you sure about that? I vaguely remember ``course-id`` being a part of the analytics event on Segment.

Comment (clintonb) [reply ecommerce/templates/oscar/basket/partials/client_side_checkout_basket.html]: Show me. We have no code that accesses this attribute. I have found no references in Google Tag Manager.

Comment (clintonb) [reply ecommerce/templates/oscar/basket/partials/client_side_checkout_basket.html]: I any case, this is not the way we should be adding properties to events. Just as we should never assume a basket contains one item, we should never assume it contains a specific type of product.

**Comment by marjev** [comment]
I removed `data-course-id` attribute and `course_key` context variable.
Everything else is back.

**Review by clintonb — APPROVED** [review]
Squash your commits.

**Comment by marjev** [comment]
@vkaracic thumbs?

**Review by vkaracic — APPROVED** [review]

**Pull request merged**